share/completions/sops.fish

```diff
@@ -0,0 +1,147 @@
+# Completion for SOPS (Secrets OPerationS)
+
+function __fish_sops_commands --description "Test if argument(s) match a sops command"
+  if not set -q argv[1]
+    return 1
+  end
+  set -l cmd (commandline -cp)
+  if set -q argv[2]
+    if string match -qr -- "sops\s+$argv[1]\s+$argv[2]\s+\S*" $cmd
+      return 0
+    end
+    return 1
+  end
+  if string match -qr -- "sops\s+$argv[1]\s+\S*" $cmd
+    return 0
+  end
+  return 1
+end
+
+# Primary commands
+complete -F -c sops -n __fish_is_first_arg -a exec-env -d "execute a command with decrypted values inserted into the environment"
+complete -F -c sops -n __fish_is_first_arg -a exec-file -d "execute a command with decrypted contents as a temporary file"
+complete -F -c sops -n __fish_is_first_arg -a publish -d "publish sops file or directory to a configured destination"
+complete -F -c sops -n __fish_is_first_arg -a keyservice -d "start a SOPS key kervice server"
+complete -F -c sops -n __fish_is_first_arg -a groups -d "modify the groups on a SOPS file"
+complete -F -c sops -n __fish_is_first_arg -a updatekeys -d "update the keys of a SOPS file using the config file"
+complete -F -c sops -n __fish_is_first_arg -a help -d "shows a list of commands or help for one command"
+complete -F -c sops -n __fish_is_first_arg -a h -d "shows a list of commands or help for one command"
+
+# Primary flags without parameters
+complete -F -c sops -n __fish_is_first_token -s d -l decrypt -d "decrypt a file and output the result to stdout"
+complete -F -c sops -n __fish_is_first_token -s e -l encrypt -d "encrypt a file and output the result to stdout"
+complete -F -c sops -n __fish_is_first_token -s r -l rotate -d "generate a new data encryption key and reencrypt all values with the new key"
+complete -F -c sops -n __fish_is_first_token -s i -l in-place -d "write output back to the same file instead of stdout"
+complete -F -c sops -n __fish_is_first_token -s s -l show-master-keys -d "display master encryption keys in the file during editing"
+complete -F -c sops -n __fish_is_first_token -l extract -d "extract a specific key or branch from the input document. Decrypt mode only. Example: --extract '[\"somekey\"][0]'"
```

Comment:
The description is probably too long and might be cut off, so how about

```suggestion
complete -F -c sops -n __fish_is_first_token -l extract -d "extract a specific key or branch from the decrypted input document"
```

same for some other descriptions. I think 60 chars is our limit. They are summaries, not documentation.